File: lite-app-starter/src/main/java/com/chimbori/liteapp/starter/HermitIntentStarter.java
package com.chimbori.liteapp.starter;

import android.app.Activity;
import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;

import androidx.annotation.NonNull;

public class HermitIntentStarter {
  @NonNull
  public static Intent createLiteAppIntent(@NonNull String startUrl) {
    Uri startUri = Uri.parse(startUrl);
    return new Intent(Intent.ACTION_VIEW, startUri)
        .setClassName(Constants.HERMIT_PACKAGE_NAME, Constants.HERMIT_ACTIVITY_NAME);
  }

  public static void launchLiteApp(@NonNull Activity activity, @NonNull String startUrl) {
    try {
      activity.startActivity(createLiteAppIntent(startUrl));
    } catch (ActivityNotFoundException e) {
      activity.startActivity(new Intent(activity, InstallActivity.class));
    }
  }
}

File: lite-app-starter/src/main/java/com/chimbori/liteapp/starter/HermitIntentStarter.kt
package com.chimbori.liteapp.starter

import android.app.Activity
import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import com.chimbori.liteapp.starter.Constants.HERMIT_ACTIVITY_NAME
import com.chimbori.liteapp.starter.Constants.HERMIT_PACKAGE_NAME

object HermitIntentStarter {
  private fun createLiteAppIntent(startUrl: String) =
      Intent(Intent.ACTION_VIEW, Uri.parse(startUrl)).setClassName(HERMIT_PACKAGE_NAME, HERMIT_ACTIVITY_NAME)

  fun launchLiteApp(activity: Activity, startUrl: String) = try {
    activity.startActivity(createLiteAppIntent(startUrl))
  } catch (e: ActivityNotFoundException) {
    activity.startActivity(Intent(activity, InstallActivity::class.java))
  }
}